notebook/Solution/03 - Time Series Forecasting Using MLP.ipynb

```diff
@@ -0,0 +1,1693 @@
+    "# function should return features in sequence and labels\n",
+    "def sequencing_data(data, sequence_length):\n",
+    "    \"\"\"\n",
+    "    Helper function to sample sub-sequence of training data.\n",
+    "    Input data must be in np.ndarray.\n",
+    "    \"\"\"\n",
+    "    x, y = [], []\n",
+    "\n",
+    "    for i in range(data.shape[0] - sequence_length):\n",
+    "\n",
+    "        # copy the sequences of data starting at this index\n",
+    "        x.append(data[i:i + sequence_length])\n",
+    "        y.append(data[i + sequence_length])\n",
+    "    return x, y\n",
+    "\n",
+    "# calling the helper function and store them in variables\n",
+    "X_sequence_train, y_sequence_train = sequencing_data(train_scaled, 3)\n",
+    "X_sequence_test, y_sequence_test = sequencing_data(test_scaled, 3)\n",
+    "\n",
+    "# sanity check \n",
+    "print(\"Total samples for X train: \" + str(len(X_sequence_train)))\n",
+    "print(\"Total samples for y train: \" + str(len(y_sequence_train)))\n",
+    "print(\"Total samples for X test: \" + str(len(X_sequence_test)))\n",
+    "print(\"Total samples for y test: \" + str(len(y_sequence_test)))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Let's write a custom `Dataset` class to transfrom the data into tuples of `torch.Tensor`. This is necessary if we want to load the data in batches using `DataLoader`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class ShampooDataset(Dataset):\n",
+    "    \"\"\"\n",
+    "    Convert input data into torch FloatTensor. \n",
+    "    Inherit Dataset class. Return length of instance when len method is called and return specific sample\n",
+    "    of data when indexed.\n",
+    "    \"\"\" \n",
+    "    def __init__(self, x, y):\n",
+    "        self.x = torch.FloatTensor(x)\n",
+    "        self.y = torch.FloatTensor(y)\n",
+    "        \n",
+    "    def __len__(self):\n",
+    "        return self.x.size(0)\n",
+    "    \n",
+    "    def __getitem__(self, idx):\n",
+    "        return self.x[idx], self.y[idx]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We will write the codes to convert data into `Dataset` object and then into `DataLoader` object. We will be using a batch size of 4 for this `DataLoader` object."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 12,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# declare batch size\n",
+    "batch_size = 4\n",
+    "\n",
+    "# convert into Dataset using customized class\n",
+    "train_Dataset = ShampooDataset(X_sequence_train, y_sequence_train)\n",
+    "test_Dataset = ShampooDataset(X_sequence_test, y_sequence_test)\n",
+    "\n",
+    "# convert into DataLoader\n",
+    "train_loader = DataLoader(train_Dataset, batch_size=batch_size, shuffle=True)\n",
```

Comment:
should be shuffle=False because we don want to break the information for the sequence

Reply:
What's breaking sequence information if one sequence only 3 data?

Reply:
yes, your point is correct, if not mistaken, you are breaking the sequence information between batches of data.For example 
data_list - [1 2 3 4 5 ]
Batch 1 - [1 2 3]
Batch 2 - [2 3 4]
Batch 3 - [2 4 5]
if shuffle =  False, the batch data will be arranged in Batch 1->Batch 2-> Batch 3
if shuffle = True, the batch data will be arranged in Batch 1 -> Batch 3 ->Batch 2

Reply:
As per our discussion, resolving this